### PR TITLE
fix: auto apply formatter args set in `user/configs/formatters`.

### PR DESCRIPTION
**Pull request by ayamir**
This PR supports applying formatter args set in `user/configs/formatters`. The related wiki will be updated after merging.

### CODE REVIEW

**Comment by ayamir** [comment]
checkout to this PR and copy the content in `user_templated` to `user`.
Move your `clang_format.lua` to `lua/user/configs/formatters/`.
@Penguin-SAMA

**Comment by ayamir** [comment]
https://github.com/ayamir/nvimdots/issues/1048?notification_referrer_id=NT_kwDOA6zRN7M4MjUxMTE1MzMwOjYxNjU3Mzk5#issuecomment-1788781832

**Review by CharlesChiuGit — APPROVED** [review]
didn't use clang-format, so i can't test it; however, the lua part make sense to me.

File: lua/modules/configs/completion/null-ls.lua [hunk 1]
```diff
@@ -2,12 +2,26 @@ return function()
...
 	local btns = null_ls.builtins
 
+	---Return formatter args required by `extra_args`
+	---@param formatter_name string
+	---@return table|nil
+	local function formatter_args(formatter_name)
+		local ok, args = pcall(require, "user.configs.formatters." .. formatter_name)
+		if not ok then
+			ok, args = pcall(require, "completion.formatters." .. formatter_name)
+		end
```
Comment (Jint-lzxy) [inline lua/modules/configs/completion/null-ls.lua:12]: IMO we can get rid of this `pcall` b/c we do ensure that the formatting specification(s) exist in the base config.

**Review by Jint-lzxy — CHANGES_REQUESTED** [review]

**Review by Jint-lzxy — APPROVED** [review]
lgtm

**Pull request merged**